pkg/parser/additional/additional.go

```diff
@@ -0,0 +1,199 @@
+package additional
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+)
+
+func checkConvertingError(err error) {
+	if err != nil {
+		log.Error().Msg("Failed to convert 'string' into 'int'")
+	}
+}
+
+func getRegexMatch(regex, text string) string {
+	var re = regexp.MustCompile(regex)
+	clean := re.FindAllString(text, -1)
+
+	if clean == nil {
+		return ""
+	}
+
+	return clean[0]
+}
+
+func getExpirationDate(datetime string) [3]int {
+	// get notAfter datetime
+	expirationInfo := getRegexMatch(`(notAfter=)[a-zA-Z]+[ ]+[0-9]{1,2}[ ]([0-9]{2}:[0-9]{2}:[0-9]{2})[ ][0-9]{4}`, datetime)
+	clean := strings.Split(expirationInfo, "=")
+
+	months := map[string]int{
+		"Jan": 1,
+		"Feb": 2,
+		"Mar": 3,
+		"Apr": 4,
+		"May": 5,
+		"Jun": 6,
+		"Jul": 7,
+		"Aug": 8,
+		"Sep": 9,
+		"Oct": 10,
+		"Nov": 11,
+		"Dec": 12,
+	}
+
+	// pick month
+	cleanMonth := getRegexMatch(`^[a-zA-Z]+`, clean[1])
+	month := months[cleanMonth]
+
+	// pick year
+	cleanYear := getRegexMatch(`[0-9]{4}`, clean[1])
+	year, errYear := strconv.Atoi(cleanYear)
+	checkConvertingError(errYear)
+
+	// pick day
+	cleanDay := getRegexMatch(`[ ][0-9]{1,2}`, clean[1])
+	day, errDay := strconv.Atoi(strings.ReplaceAll(cleanDay, " ", ""))
+	checkConvertingError(errDay)
+
+	var date [3]int
+	date[0] = year
+	date[1] = month
+	date[2] = day
+
+	return date
+}
+
+func expirationDateFromPemCertificate(filePath string) [3]int {
+	out, err := exec.Command("openssl", "x509", "-in", filePath, "-noout", "-dates").Output()
+	if err != nil {
+		log.Error().Msgf("Failed to execute command 'openssl x509 -in %s -noout -dates'", filePath)
+	}
+
+	date := getExpirationDate(string(out))
+
+	return date
+}
+
+func getBitRSAKeyBit(filePath string) int {
+	out, err := exec.Command("openssl", "x509", "-in", filePath, "-text", "-noout").Output()
+
+	if err != nil {
+		log.Error().Msgf("Failed to execute command 'openssl x509 -in %s -text -noout'", filePath)
+		return -1
+	}
+
+	rsaInfo := getRegexMatch(`RSA Public-Key:[ ]\([0-9]{4}[ ]bit\)`, string(out))
```

Comment:
The same concern as in the previous comment.

Take a look at this example:

```go
package main

import (
	"crypto/x509"
	"encoding/pem"
	"fmt"
	"os"
)

func main() {
	certPEM, err := os.ReadFile("google-ssl.pem")
	if err != nil {
		panic("failed to find certificate")
	}

	block, _ := pem.Decode(certPEM)
	if block == nil {
		panic("failed to parse certificate PEM")
	}
	cert, err := x509.ParseCertificate(block.Bytes)
	if err != nil {
		panic("failed to parse certificate: " + err.Error())
	}
	//fmt.Println(cert.NotAfter)
	publicKeyDer, _ := x509.MarshalPKIXPublicKey(cert.PublicKey)
	publicKeyBlock := pem.Block{
		Type:  "PUBLIC KEY",
		Bytes: publicKeyDer,
	}
	publicKeyPem := string(pem.EncodeToMemory(&publicKeyBlock))
	fmt.Println(publicKeyPem)
}
```

![image](https://user-images.githubusercontent.com/75368139/115960849-8f293d80-a50b-11eb-94be-b38a796e3440.png)